packetbeat/protos/http/http.go

```diff
@@ -460,26 +460,31 @@ func (http *HTTP) newTransaction(requ, resp *message) common.MapStr {
 		src, dst = dst, src
 	}
 
-	details := common.MapStr{
-		"phrase":         resp.StatusPhrase,
-		"code":           resp.StatusCode,
-		"content_length": resp.ContentLength,
-	}
-	if http.parserConfig.SendHeaders {
-		details["request_headers"] = http.collectHeaders(requ)
-		details["response_headers"] = http.collectHeaders(resp)
+	http_details := common.MapStr{
+		"request": common.MapStr{
+			"params": params,
+		},
+		"response": common.MapStr{
+			"code":   resp.StatusCode,
+			"phrase": resp.StatusPhrase,
+		},
 	}
 
+	http_details["request"].(common.MapStr)["headers"] = http.collectHeaders(requ)
```

Comment:
You could avoid the type assertion here by moving these inside the MapStr declaration above.